infinite-tracing/src/main/java/com/newrelic/ResponseObserver.java

```diff
@@ -43,6 +46,14 @@ public void onError(Throwable t) {
             return;
         }
 
+        if (firstFailedPrecondition.get() && isFailedPrecondition(t)) {
```

Comment:
This handles the first `Failed_Precondition`, the one that should trigger an immediate reconnect